cell_classification/predict.py

```diff
@@ -0,0 +1,375 @@
+import h5py
+import toml
+import argparse
+import pickle
+import os
+from sklearn.metrics import roc_curve, auc, confusion_matrix
+from model_builder import ModelBuilder
+import numpy as np
+from tqdm import tqdm
+from plot_utils import plot_average_roc, plot_metrics_against_threshold
+import toml
+from copy import deepcopy
+from joblib import Parallel, delayed
+import pandas as pd
+import tables
+
+
+def load_model_and_val_data(params):
+    """Load model and validation data from params dict
+    Args:
+        params (dict):
+            dictionary containing model and validation data paths
+    Returns:
+        model (ModelBuilder):
+            trained model
+        val_data (tf.data.Dataset):
+            validation dataset
+    """
+    params["eval"] = True
+    model = ModelBuilder(params)
+    model.prep_data()
+    model.load_model(params["model_path"])
+    val_dset = iter(model.validation_dataset)
+    return model, val_dset
+
+
+def process_to_cells(sample):
+    """Process predictions from pixel level to cell level
+    Args:
+        sample (dict):
+            single sample with predictions
+    Returns:
+        sample (dict):
+            single sample with predictions at cell level
+    """
+    if not isinstance(sample["activity_df"], pd.DataFrame):
+        if isinstance(sample["activity_df"], str):
+            df = pd.read_json(sample["activity_df"])
+        else:
+            df = pd.read_hdf(sample["activity_df"])
+    df["pred_activity"] = 0.0
+    unique_labels = np.unique(sample["instance_mask"])
+    unique_labels = unique_labels[unique_labels != 0]
+    mean_per_cell_mask = np.zeros_like(sample["instance_mask"], dtype=np.float32)
+    for unique_label in unique_labels:
+        mask = sample["instance_mask"] == unique_label
+        mean_pred = sample["prediction"][mask].mean()
+        df.at[df.index[df["labels"] == unique_label].to_list().pop(), "pred_activity"] = mean_pred
+        mean_per_cell_mask[mask] = mean_pred
+    sample["activity_df"] = df
+    sample["prediction_mean"] = mean_per_cell_mask
+    return sample
+
+
+def predict(model, val_dset, params, return_pred=False):
```

Comment:
Tons of new code, which is great. Let's talk about it tomorrow after standup, I think it'll be easier to go through this in person. 

The main thing I want to discuss: what data structures are we using for each step of the model evaluation pipeline?